sonar-plugin-api/src/main/java/org/sonar/api/measures/MetricDefinition.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.api.measures;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.CheckForNull;
+import org.sonar.api.config.Settings;
+
+/**
+ * TODO
+ *
+ *
+ */
+public interface MetricDefinition {
+
+  class NewMetricContext {
+    private final Map<String, Metric> metricsByKey = Maps.newHashMap();
+
+    public NewMetric createNewMetric() {
+      return new NewMetricImpl(this);
+    }
+
+    @CheckForNull
+    public Metric getMetric(String metricKey) {
+      return metricsByKey.get(metricKey);
+    }
+
+    public List<Metric> getMetrics(){
+      return ImmutableList.copyOf(metricsByKey.values());
+    }
+
+    private void registerMetric(NewMetricImpl newMetric) {
+      // TODO check metric does not already exists
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)